### PR TITLE
Icu 10504 use hds table in desktop sessions list

### PR DESCRIPTION
**Pull request by NatashaKab**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JICU-10504)

## Description

:technologist: [Admin preview](https://boundary-ui-desktop-git-icu-10504-use-hds-tabl-da8730-hashicorp.vercel.app/scopes/global/projects/targets)

:rose: [Rose preview](https://boundary-ui-storybook.vercel.app/?path=/story/rose-table--basic)

### Screenshots (if appropriate):
Before:
Screenshot 2023-08-03 at 3 18 29 PM

After:
Screenshot 2023-08-03 at 5 24 12 PM

### CODE REVIEW

**Review by cameronperera — APPROVED** [review]
LGTM!

**Comment by joshuaogle** [comment]
This looks so nice!! 👏

**Review by DhariniJeeva — APPROVED** [review]
Great job! 💯

File: ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs [hunk 1]
```diff
@@ -115,80 +115,83 @@
...
-                  Two if's here because:
-                  1) Local proxy info is only relevant if the session is
-                     active or pending (aka isCancelable).  Once the session
-                     is canceled or in the process thereof, the local proxy
-                     may no longer be used.
-                  2) The session actually has proxy information associated
-                     with it.  It's possible for the user to start sessions
-                     from other devices (or even another tab), in which case we
-                     won't have the local proxy info available here.
-                }}
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs:158]: Any reason this comment was deleted? Is it still relevant, if so it seemed pretty informational

File: ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs [hunk 2]
```diff
@@ -115,80 +115,83 @@
...
+          <B.Tr>
+            <B.Td>
+              {{#if (can 'read model' B.data)}}
+                {{#if B.data.isActive}}
+                  <Hds::Badge
+                    @text={{t 'resources.session.title'}}
+                    @icon='entry-point'
+                    @isIconOnly={{true}}
+                    @color='success'
+                    class='session-status-active'
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs:140]: Is this class still needed? It looks like it only sets the color which is handled by the new `@color` field

Comment (cameronperera) [reply ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs:140]: Good catch, I forgot to comment on this class. We have a test that uses that class to verify there is an active session. Looking at the designs for this table, it is going to change a bit and that icon and test will go away. To fight off scope creep, I told Natasha to add the class for now until we refactor the table items for the post connection workflow updates.

Comment (ZedLi) [reply ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs:140]: We should add a TODO then to note this

File: ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs [hunk 3]
```diff
@@ -115,37 +115,53 @@
...
+              {{#if (can 'read model' B.data)}}
+                {{#if B.data.isActive}}
+                  <Hds::Badge
+                    @text={{t 'resources.session.title'}}
+                    @icon='entry-point'
+                    @isIconOnly={{true}}
+                    @color='success'
+                    {{! there are plans to remove this class. It is here to make the test 
+                    pass. This is going to change later so Cameron suggested it's ok to
+                    have it for now.}}
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/sessions/index.hbs:142]:
```suggestion
                    {{! TODO: Remove this class once the table gets refactored. 
                    It's not currently needed but is only used for tests to pass. }}
```

Small suggestion to actually use a `TODO` (which makes it easier to find these comments)

**Review by ZedLi — APPROVED** [review]
Looks good!

**Pull request merged**